conjure-generator-common/src/test/java/com/palantir/conjure/PatternTestUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure;
+
+import java.util.List;
+import java.util.Random;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public final class PatternTestUtil {
```

Comment:
It might be worthwhile to borrow from https://github.com/palantir/resource-identifier/blob/develop/src/test/java/com/palantir/ri/ResourceIdentifierPropertyTest.java

Reply:
Inversely, might be worth writing hand rolled optimized RID parser/validator

Reply:
That's exactly what that test is validating :-) It's ensuring that the old regex agrees with the new hand-rolled validator

Contributed here: https://github.com/palantir/resource-identifier/pull/140/

Reply:
Doh! I missed that PR landing for RIDs, it had popped up in some historic perf profiles I was looking at a while back. I know there are a few other places doing RID-like parsing that have popped in profiling that could use similar love.

Reply:
Please let us know when those come up, happy to look into improving them :-)